Refresh the repo README (#10481)
hashicorp/terraform-provider-google: README.md

==================

> - Website: https://www.terraform.io

Reviewer: This is the website for terraform, but this provider isn't on it. Instead of doing this here, I actually wonder if you could do an in-text link in the initial description like "The google provider is a plugin that allows [Terraform](https://www.terraform.io) to manage infrastructure on Google Cloud Platform."

- Tutorials: [learn.hashicorp.com](https://learn.hashicorp.com/terraform?track=getting-started#getting-started)
- Forum: [discuss.hashicorp.com](https://discuss.hashicorp.com/c/terraform-providers/tf-google/)
- Documentation: https://www.terraform.io/docs/providers/google/index.html
- [![Gitter chat](https://badges.gitter.im/hashicorp-terraform/Lobby.png)](https://gitter.im/hashicorp-terraform/Lobby)
- Mailing list: [Google Groups](http://groups.google.com/group/terraform-tool)

> The Terraform Google provider is a plugin for Terraform that allows for management of GCP resources.
> This provider is maintained by the [Terraform team at Google](https://cloudplatform.googleblog.com/2017/03/partnering-on-open-source-Google-and-HashiCorp-engineers-on-managing-GCP-infrastructure.html) and the Terraform team at [HashiCorp](https://www.hashicorp.com/)

Reviewer: ```suggestion
The Terraform Google provider is a plugin that allows Terraform to manage resources on Google Cloud Platform. This provider is maintained by the [Terraform team at Google](https://cloudplatform.googleblog.com/2017/03/partnering-on-open-source-Google-and-HashiCorp-engineers-on-managing-GCP-infrastructure.html) and the Terraform team at [HashiCorp](https://www.hashicorp.com/)
```

> Also see the ['google-beta' provider](https://github.com/hashicorp/terraform-provider-google-beta) for preview features and features at a beta [launch stage](https://cloud.google.com/products#product-launch-stages). See [Provider Versions](https://www.terraform.io/docs/providers/google/provider_versions.html) for more details on how to use `google-beta`.

Reviewer: I also think you should start the README off with this nice description of what this is :-)  Then I think you can move into the get started section and any links.  I just think if folks come here, the first thing they'd want is to understand what the google provider is.

```suggestion
The ['google-beta' provider](https://github.com/hashicorp/terraform-provider-google-beta) contains preview features and features at a beta [launch stage](https://cloud.google.com/products#product-launch-stages). Refer to the [provider versions](https://www.terraform.io/docs/providers/google/provider_versions.html) documentation for more information about how to use `google-beta`.
```

> Quick Starts

Reviewer: Okay I think you could make this a true get started section and have the get started quick start you have linked plus maybe the tutorials and documentation link?  Then I wonder if it might be nice to have a "Community" or a "Contribute" section where you link the community forum and provide a link to the instructions about provider development?

```suggestion
Get Started
```

----------------------

- [Getting Started with the Google Provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/getting_started)
- [Provider Development](docs/contributing)

> Documentation

Reviewer: Do we need this section? I only ask because I see we have documentation linked above already!
